dotnet-sdk-80/snapcraft.yaml

```
@@ -47,6 +47,54 @@ parts:
         craftctl set version="${DOTNET_SDK_VERSION}"
       fi
 
+  mounts:
+    plugin: dump
+    source: .
+    after:
+      - dotnet-sdk
+    override-build: |
+      RUNTIME_VERSION=$(ls ${CRAFT_STAGE}/usr/lib/dotnet/shared/Microsoft.NETCore.App/ | grep 8.0)
+      SDK_VERSION=$(ls ${CRAFT_STAGE}/usr/lib/dotnet/sdk | grep 8.0)
+      
+      if [ "${CRAFT_ARCH_BUILD_FOR}" = "amd64" ]; then
+        ARCH="x64"
+      elif [ "${CRAFT_ARCH_BUILD_FOR}" = "arm64" ]; then
+        ARCH="arm64"
+      else
+        echo "Unknown architecture (${CRAFT_ARCH_BUILD_FOR})"
+        exit 1
+      fi
+
+      for file in mounts/*; do
+        sed -i "s/{RT_VERSION}/$RUNTIME_VERSION/g" $file
+        sed -i "s/{SDK_VERSION}/$SDK_VERSION/g" $file
+        sed -i "s/{SNAP}/$CRAFT_PROJECT_NAME/g" $file
+        sed -i "s/{ARCH}/$ARCH/g" $file
+
+        if [[ "$file" == *"{RT_VERSION}"* ]]; then
+          new_file=$(echo "$file" | sed "s/{RT_VERSION}/$RUNTIME_VERSION/")
+          mv "$file" "$new_file"
+          file=$new_file
+          echo "Renamed '$file' to '$new_file'"
+        fi
+        if [[ "$file" == *"{SDK_VERSION}"* ]]; then
+          new_file=$(echo "$file" | sed "s/{SDK_VERSION}/$SDK_VERSION/")
+          mv "$file" "$new_file"
+          file=$new_file
+          echo "Renamed '$file' to '$new_file'"
+        fi
+        if [[ "$file" == *"{ARCH}"* ]]; then
+          new_file=$(echo "$file" | sed "s/{ARCH}/$ARCH/")
+          mv "$file" "$new_file"
+          file=$new_file
+          echo "Renamed '$file' to '$new_file'"
+        fi
```

suggestion: I think this can be simplified
```suggestion
        new_file_name="$(echo "$file" \
          | sed "s/{RT_VERSION}/$RUNTIME_VERSION/" \
          | sed "s/{SDK_VERSION}/$SDK_VERSION/" \
          | sed "s/{ARCH}/$ARCH/")"
        
        if [[ "$file" != "$new_file_name" ]]; then  
          mv "$file" "$new_file_name"
          echo "Renamed '$file' to '$new_file'"
          file="$new_file_name"
        fi
```

similar for the other `snapcraft.yaml` files too

---

note: this also has the benefit that you do not rename and log one file twice e.g. for `dotnet-sdk-80/mounts/var-snap-dotnet-common-dotnet-packs-Microsoft.NETCore.App.Runtime.ubuntu.22.04-{ARCH}-{RT_VERSION}.mount`

---

Thank you! This is much better.